### PR TITLE
skip over empty fastqs after demultiplex

### PR DESCRIPTION
**Pull request by glichtenstein**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/demultiplex/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/demultiplex _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: subworkflows/nf-core/bcl_demultiplex/main.nf [hunk 1]
```diff
@@ -72,6 +75,18 @@ workflow BCL_DEMULTIPLEX {
...
+    output:
+    file("invalid_fastqs.txt") into ch_failed_fastqs_file
+
+    publishDir "${params.outdir}", mode: 'copy'
+
+    script:
+    """
+    cat ${invalid_fastqs_ch.collect().join('\n')} > invalid_fastqs.txt
+    """
+}
```
Comment (maxulysse) [inline subworkflows/nf-core/bcl_demultiplex/main.nf:88]: I think we can do that outside a process with a collectFile operator

Comment (edmundmiller) [reply subworkflows/nf-core/bcl_demultiplex/main.nf:88]: Agreed, that's one way to do it, but I think it can just be native Nextflow.

**Comment by glichtenstein** [comment]
I will work on this more, I still need to do more tests. I'm considering making that behaviour for skipping emtpy fastqs optional. Should I close the pull request or can I leave it as a draft? Thanks for the reviews.

**Comment by Aratz** [comment]
I'd leave it as draft, so that everybody can see what the status is with this issue and maybe not start something else from scratch.

**Pull request closed without merging**